diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbf0999ce8c66c886196781b3f3deb7204a3f30e..6d74709a3e93bdb0867db20bc349cd1e04960d37 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -538,6 +538,7 @@ if (NOT CMAKE_VERSION VERSION_LESS 3.0)

   # Imported target support
   add_library (eigen INTERFACE)
+  add_library (Eigen3::Eigen ALIAS eigen)

   target_compile_definitions (eigen INTERFACE ${EIGEN_DEFINITIONS})
   target_include_directories (eigen INTERFACE
diff --git a/cmake/Eigen3Config.cmake.in b/cmake/Eigen3Config.cmake.in
index c5c54688771aae50c8c8bbb79b90f07b718a2757..0a1ac61c94d79fa917d7483d14b511085db7319a 100644
--- a/cmake/Eigen3Config.cmake.in
+++ b/cmake/Eigen3Config.cmake.in
@@ -3,7 +3,9 @@

 @PACKAGE_INIT@

-include ("${CMAKE_CURRENT_LIST_DIR}/Eigen3Targets.cmake")
+if (NOT TARGET eigen)
+  include ("${CMAKE_CURRENT_LIST_DIR}/Eigen3Targets.cmake")
+endif ()

 # Legacy variables, do *not* use. May be removed in the future.
